codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -404,12 +415,12 @@ private void applyLocalSettings(Http2Settings settings) throws Http2Exception {
 
             Long headerTableSize = settings.headerTableSize();
             if (headerTableSize != null) {
-                headerTable.maxHeaderTableSize(headerTableSize);
+                headerConfig.maxHeaderTableSize(headerTableSize);
             }
 
             Long maxHeaderListSize = settings.maxHeaderListSize();
             if (maxHeaderListSize != null) {
-                headerTable.maxHeaderListSize(maxHeaderListSize);
+                headerConfig.maxHeaderListSize(maxHeaderListSize, maxHeaderListSizeGoAway(maxHeaderListSize));
```

Comment:
I don't think so, but maybe I don't understand what you are asking. We enforce 2 limits:

1. maxHeadersListSize - This directly correlates to SETTINGS_MAX_HEADER_LIST_SIZE. We choose to interpret this limit as "send a RST_STREAM if it is exceeded.". In order to send a RST_STREAM we have to still process the entire header block, or HPACK will be out of sync (this is why the second limit is necessary).
2. maxHeadersListSizeGoAway - Derived from maxHeadersListSize and provides an upper bound as to how much data we will tolerate before we throw our hands up and send a GO_AWAY.

Both of these limits are derived from the last value we communicated to our peer.